shared/util.go

```diff
@@ -1371,3 +1371,18 @@ func JoinTokenDecode(input string) (*api.ClusterMemberJoinToken, error) {
 
 	return &j, nil
 }
+
+// TargetDetect returns either target node or group based on the provided prefix:
+// An invocation with `target=h1` returns "h1", "" and
+// `target=@g1` returns "", "g1".
+func TargetDetect(target string) (string, string) {
```

Comment:
Given you're defining these vars at top of function, using named return vars in function definition would achieve the same effect but would be clearer to the reader in my view.

Reply:
Changed to named return vars.